pennylane/templates/layers/basic_entangler.py

```diff
@@ -157,14 +122,91 @@ def circuit(weights):
         ``ValueError: Wrong number of parameters``.
     """
 
-    if rotation is None:
-        rotation = RX
+    num_params = 1
+    num_wires = AnyWires
+    par_domain = "A"
+
+    def __init__(self, weights, wires=None, rotation=None, do_queue=True):
+
+        self.rotation = rotation or qml.RX
+
+        super().__init__(weights, wires=wires, do_queue=do_queue)
+        self._preprocess()
+
+    def expand(self):
```

Comment:
I really like `expand()`! It's a clever approach for writing complicated operations